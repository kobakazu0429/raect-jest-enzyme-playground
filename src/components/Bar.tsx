import React, { useState, useCallback, ChangeEvent } from "react";

export interface IProps {
  name: string;
  age?: number;
}

export const Bar = ({ name, age: _age = 0 }: IProps) => {
  const [age, setAge] = useState(_age);
  const handleChange = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      setAge(Number(e.target.value));
    },
    [age]
  );

  return (
    <div>
      <p id="name">name: {name}</p>
      <p id="age">age: {age}</p>
      <input type="number" value={age} onChange={handleChange} />
    </div>
  );
};
