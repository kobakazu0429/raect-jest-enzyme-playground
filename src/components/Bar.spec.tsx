import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import { Bar, IProps } from "@/components/Bar";

const initialProps: IProps = { name: "Taro", age: 10 };

describe("<Bar />", () => {
  const wrapper = shallow(<Bar {...initialProps} />);
  it("pタグが2つある", () => {
    expect(wrapper.find("p").length).toBe(2);
  });

  it("inputタグが1つある", () => {
    expect(wrapper.find("input").length).toBe(1);
  });

  it("#name with initialProps", () => {
    expect(wrapper.find("#name").text()).toBe(`name: ${initialProps.name}`);
  });

  it("#age with initialProps", () => {
    expect(wrapper.find("#age").text()).toBe(`age: ${initialProps.age}`);
  });

  it("#age with setAge()", () => {
    expect(wrapper.find("#age").text()).toBe(`age: ${initialProps.age}`);

    const newAge = 21;
    wrapper.find("input").simulate("change", {
      target: {
        value: newAge
      }
    });
    expect(wrapper.find("#age").text()).toBe(`age: ${newAge}`);
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<Bar {...initialProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
