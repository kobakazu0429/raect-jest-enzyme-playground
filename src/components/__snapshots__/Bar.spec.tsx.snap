// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar /> should match the snapshot 1`] = `
<div>
  <p
    id="name"
  >
    name: 
    Taro
  </p>
  <p
    id="age"
  >
    age: 
    10
  </p>
  <input
    onChange={[Function]}
    type="number"
    value={10}
  />
</div>
`;
